src/Data/ProgrammesDb/EntityRepository/BroadcastRepository.php

```diff
@@ -98,4 +99,48 @@ private function setEntityTypeFilter(QueryBuilder $qb, string $type, string $bro
 
         return $qb;
     }
+
+    public function findAllByServiceAndDateRange(
+        Sid $serviceId,
+        DateTimeImmutable $startDate,
+        DateTimeImmutable $endDate,
+        ?int $limit,
```

Comment:
well, "?int ... " means that can be "int" but also "null", but no optional. 
Also our AbstractService::NO_LIMIT has the limit null as default.
https://github.com/bbc/programmes-pages-service/blob/master/src/Service/AbstractService.php#L14


Not sure about if we should change this. because Look at this offset:
https://github.com/bbc/programmes-pages-service/blob/master/src/Service/AbstractService.php#L47

I will ask around.